### PR TITLE
fix: impure rust unit tests

### PR DESCRIPTION
**Pull request by ysndr**
Fix failing rust unit tests and enable running them in CI

### CODE REVIEW

**Comment by ysndr** [comment]
@zmitchell: tagged you as reviewer since i disaled the tests for install/uninstall. Just so youre aware to remove the `#[ignore]` when you get to fix these tests

File: crates/flox-rust-sdk/src/models/publish.rs [hunk 1]
```diff
@@ -968,6 +969,7 @@ mod tests {
 
     #[cfg(feature = "impure-unit-tests")] // /nix/store is not accessible in the sandbox
     #[tokio::test]
+    #[ignore = "disabled as `publish` is deprecated"]
```
Comment (zmitchell) [inline crates/flox-rust-sdk/src/models/publish.rs:972]: Should we not just delete the publish tests?

File: crates/flox-rust-sdk/src/models/publish.rs [hunk 2]
```diff
@@ -938,6 +938,7 @@ pub enum ConvertFlakeRefError {
 }
 
 #[cfg(test)]
+
```
Comment (zmitchell) [inline crates/flox-rust-sdk/src/models/publish.rs:941]: I don't think there's supposed to be a blank line here.

Comment (ysndr) [reply crates/flox-rust-sdk/src/models/publish.rs:941]:
```suggestion
```

File: crates/flox-rust-sdk/src/models/publish.rs (hunk 1 above)
Comment (ysndr) [reply crates/flox-rust-sdk/src/models/publish.rs:971]: Id rather remove the entire publish as a separate commit/PR.
That way we can pull it back in as a whole when we pick up the work on it.

**Review by zmitchell — APPROVED** [review]

**Pull request merged**